jsonnet/prometheus-adapter.libsonnet

```diff
@@ -69,8 +70,27 @@ function(params)
       {
         spec+: {
           replicas: 2,
+          strategy+: {
+            // Apply HA conventions
+            rollingUpdate: {
+              maxUnavailable: '25%',
+            },
+          },
           template+: {
             spec+: {
+              affinity+: {
+                podAntiAffinity+: {
+                  // Apply HA conventons
+                  requiredDuringSchedulingIgnoredDuringExecution: [
+                    {
+                      labelSelector: {
+                        matchLabels: pa.config.selectorLabels,
+                      },
+                      topologyKey: "kubernetes.io/hostname",
+                    },
+                  ],
+                },
+              },
```

Comment:
Should we add the same to kube-prometheus?

Reply:
I guess we can always make it configurable in kube-promehteus, although we are not running prometheus-adapter in HA mode by default there.

Should we add it to the [anti-affinity addon](https://github.com/prometheus-operator/kube-prometheus/blob/1d2a0e275af97948667777739a18b24464480dc8/jsonnet/kube-prometheus/addons/anti-affinity.libsonnet) or directly to the prometheus-adapter component?

Reply:
> we are not running prometheus-adapter in HA mode by default there.

Maybe we should change it then? :)

> Should we add it to the anti-affinity addon or directly to the prometheus-adapter component?

I think adding it to the addon makes the most sense.

Reply:
Sounds good I'll work on an upstream change

Reply:
https://github.com/prometheus-operator/kube-prometheus/pull/1095